include/boost/numeric/ublas/tensor/static_tensor.hpp

```diff
@@ -0,0 +1,271 @@
+        }
+
+        constexpr static_tensor( matrix_type const& v )
+        {
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : invalid extents size");
+            
+            if( !( super_type::size(0) == v.size1() || super_type::size(1) == v.size2() ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+            std::copy(v.data().begin(), v.data().end(),super_type::data_.begin());
+        }
+
+        constexpr static_tensor( matrix_type && v )
+        {
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : invalid extents size");
+            
+            auto const sz = v.size1() * v.size2();
+            
+            if( !( super_type::size(0) == v.size1() || super_type::size(1) == v.size2() ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(matrix_type &&v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+
+            for(auto i = size_type{}; i < sz; ++i){
+                super_type::data_[i] = std::move(v.data()[i]);
+            }
+        }
+
+        constexpr static_tensor (const vector_type &v){
+            
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const vector_type &v)"
+                                        " : invalid extents size");
+            
+            if( !( super_type::size(0) == v.size() || super_type::size(1) == 1 ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(const vector_type &v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+
+            std::copy(v.data().begin(), v.data().end(),super_type::data_.begin());
+            
+        }
+
+        constexpr static_tensor (vector_type &&v){
+            
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const vector_type &&v)"
+                                        " : invalid extents size");
+            
+            auto const sz = v.size();
+            
+            if( !( super_type::size(0) == v.size() || super_type::size(1) == 1 ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(vector_type &&v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+
+            for(auto i = size_type{}; i < sz; ++i){
+                super_type::data_[i] = std::move(v.data()[i]);
+            }
```

Comment:
Instead of a `for` loop, I recommend to use `std::move` algorithm.

Reply:
done.